// react-share-social/dist/App.js
"use strict";
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
var jsx_runtime_1 = require("react/jsx-runtime");
// import logo from './logo.svg';
require("./App.css");
var ShareSocial_1 = __importDefault(require("./components/ShareSocial"));
var styles_1 = require("@material-ui/core/styles");
var useStyles = (0, styles_1.makeStyles)({
    App: {
        backgroundColor: '#282c34',
        color: 'white',
        minHeight: '100vh',
        flexDirection: 'column',
        padding: '25px'
    },
    AppHeader: {
        fontStyle: 'normal',
        fontWeight: 800,
        fontSize: '2.25rem',
        lineHeight: 1.2,
        maxWidth: '933px'
    },
    AppSubHeader: {
        fontSize: '1.125rem',
        lineHeight: 1.2,
        paddingBottom: '16px',
        fontWeight: 400
    },
    Box: {
        paddingBottom: '40px'
    }
});
var style = {
    root: {
        background: 'linear-gradient(45deg, #FE6B8B 30%, #FF8E53 90%)',
        borderRadius: 3,
        border: 0,
        boxShadow: '0 3px 5px 2px rgba(255, 105, 135, .3)',
        color: 'white',
    },
    copyContainer: {
        border: '1px solid blue',
        background: 'rgb(0,0,0,0.7)'
    }
};
function App() {
    var classes = useStyles();
    return ((0, jsx_runtime_1.jsxs)("div", __assign({ className: classes.App }, { children: [(0, jsx_runtime_1.jsxs)("div", __assign({ className: classes.Box }, { children: [(0, jsx_runtime_1.jsx)("header", __assign({ className: classes.AppHeader }, { children: "Multiple social icons" })), (0, jsx_runtime_1.jsx)("h1", __assign({ className: classes.AppSubHeader }, { children: "Each Social Icon triggers a callback" })), (0, jsx_runtime_1.jsx)(ShareSocial_1.default, { socialTypes: ['facebook', 'twitter', 'line', 'linkedin', 'whatsapp', 'viber', 'telegram', 'reddit', 'instapaper', 'livejournal', 'mailru', 'ok', 'hatena',
                            'email', 'workspace'], url: "http://sifatul.github.io/", onSocialButtonClicked: function (buttonName) {
                            console.log("".concat(buttonName, " clicked"));
                        }, title: "Share and get amazing products" })] })), (0, jsx_runtime_1.jsxs)("div", __assign({ className: classes.Box }, { children: [(0, jsx_runtime_1.jsx)("header", __assign({ className: classes.AppHeader }, { children: "Everything is customizable" })), (0, jsx_runtime_1.jsx)("h1", __assign({ className: classes.AppSubHeader }, { children: "Custom styles applicable to the all parts of the UI" })), (0, jsx_runtime_1.jsx)(ShareSocial_1.default, { socialTypes: ['facebook', 'twitter', 'line', 'linkedin', 'whatsapp', 'viber', 'telegram'], url: "http://sifatul.github.io/", title: "Share and get amazing products", onSocialButtonClicked: function (buttonName) {
                            alert("".concat(buttonName, " clicked"));
                        }, style: style })] }))] })));
}
exports.default = App;
